### ISSUE TITLE
cannot compile with depthsense driver

### ISSUE THREAD

**Issue by humanoid2050** [stevenlovegrove/Pangolin#227]
When compiling, I receive the following error:

`
[ 71%] Built target pangolin
[ 72%] Linking CXX executable Testlog
../../src/libpangolin.so: undefined reference to `DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::setValue(DepthSense::Interface, char const*) const'
../../src/libpangolin.so: undefined reference to `DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::getValue_p(DepthSense::Interface) const'
../../src/libpangolin.so: undefined reference to `DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::~Property()'
../../src/libpangolin.so: undefined reference to `bool DepthSense::PropertyBase::is<DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >() const'
../../src/libpangolin.so: undefined reference to `DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > DepthSense::PropertyBase::as<DepthSense::Property<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >() const'
collect2: error: ld returned 1 exit status
test/log/CMakeFiles/Testlog.dir/build.make:122: recipe for target 'test/log/Testlog' failed
make[2]: *** [test/log/Testlog] Error 1
CMakeFiles/Makefile2:208: recipe for target 'test/log/CMakeFiles/Testlog.dir/all' failed
make[1]: *** [test/log/CMakeFiles/Testlog.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
`

What version of the DepthSense driver are you working against? I have 1.9.0-5.

I also had a separate issue which was solved by adding the following line at the beginning of src/video/drivers/depthsense.cpp

#include <pangolin/video/iostream_operators.h>

**Comment by stevenlovegrove** [comment]
This is a third-party addition to Pangolin. @jstraub does this still build for you?

**Comment by jstraub** [comment]
I have been on an old version of Pangolin and for a while now. 

Just to confirm: You are using librealsense from https://github.com/IntelRealSense/librealsense right?

**Comment by humanoid2050** [comment]
@jstraub No, this is the DepthSense driver for SoftKinetic based devices.

**Comment by stevenlovegrove** [comment]
oh, oops. @jstraub didn't write the DepthSense video addition. I haven't touched that in a while and can't help you out, sorry.
